### ISSUE TITLE
terraform corrupted state: with Error: rpc error: code = Unavailable desc = transport is closing 

### ISSUE THREAD

**Issue by poligraph** [microsoft/terraform-provider-azuredevops#192]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and Azure DevOps Provider) Version

terraform -v
Terraform v0.12.29
+ provider.azuredevops v0.0.1
+ provider.azurerm v2.28.0
+ provider.google v3.11.0

### Affected Resource(s)

* `azuredevops_branch_policy_build_validation.build-validation`

### Terraform Configuration Files

```hcl
resource "azuredevops_git_repository" "repo" {
  project_id = var.project-id
  name       = var.repo-name
  initialization {
    init_type = "Clean"
  }

  lifecycle {
    prevent_destroy = true
  }

}

resource "azuredevops_build_definition" "build" {
  count      = var.create-master-pipeline ? 1 : 0
  project_id = var.project-id
  name       = var.pipeline-name != "false" ? var.pipeline-name : var.repo-name
  path       = "\\${var.repo-folder}"

  ci_trigger {
    use_yaml = true
  }

  repository {
    repo_type = "TfsGit"
    repo_id   = azuredevops_git_repository.repo.id
    yml_path  = "pipeline.yml"
  }
}

resource "azuredevops_branch_policy_build_validation" "build-validation" {
  count      = var.enable-pr-pipeline ? 1 : 0
  project_id = var.project-id

  enabled  = var.enable-pr-pipeline
  blocking = true

  settings {
    display_name        = "check-pr"
    build_definition_id = azuredevops_build_definition.build[count.index].id
    valid_duration      = 720

    scope {
      repository_id  = azuredevops_git_repository.repo.id
      repository_ref = azuredevops_git_repository.repo.default_branch
      match_type     = "Exact"
    }
  }
}
```

### Debug Output

https://gist.github.com/poligraph/7fdec0ec3bd43118d86b5d1d83ddc3e0

### Expected Behavior

terraform plan should work

### Actual Behavior

```
Error: rpc error: code = Unavailable desc = transport is closing
Error: rpc error: code = Canceled desc = context canceled
Error: rpc error: code = Unavailable desc = transport is closing
Error: rpc error: code = Canceled desc = context canceled
```
### Steps to Reproduce

1. `terraform apply`

### Important Factoids

I have created module to create multiple repos, but second time I have got this error. Last time I recovered from previous state version, but now I cannot do that and my configuration unusable

### References

* #0000

**Comment by poligraph** [comment]
Seems like related issue I got when try to import build definitions

```
terraform import module.private_module.azuredevops_build_definition.additional-pipeline[0] 909ef74a-6216-xxxxxxxx/63
module.private_module.azuredevops_build_definition.additional-pipeline[0]: Importing from ID "909ef74a-6216-xxxxxxxx/63"...
module.private-module.azuredevops_build_definition.additional-pipeline[0]: Import prepared!
  Prepared azuredevops_build_definition for import
module.private-module.azuredevops_build_definition.additional-pipeline[0]: Refreshing state... [id=63]

Error: rpc error: code = Unavailable desc = transport is closing
```


https://gist.github.com/poligraph/4ebfd23935a8ac1006edbf855fc31f35

**Comment by xuzhang3** [comment]
@poligraph  Does your pipeline contains secret variables? If it is , this is a know issue, service will always return secret variables in empty string. I have put it in our backlog.

**Comment by poligraph** [comment]
@xuzhang3 yes, pipeline contain secrets. I see PR - will wait for a fix

**Comment by xuzhang3** [comment]
Close this issue. Feel free to reopen it if you have questions.

**Comment by poligraph** [comment]
looks like it fixed in v0.1.0.
